miio/viomivacuum.py

```diff
@@ -277,46 +281,216 @@ def mop_mode(self) -> ViomiMode:
         """
         return ViomiMode(self.data["is_mop"])
 
+    @property
+    def current_map_id(self) -> float:
+        """Current map id."""
+        return self.data["cur_mapid"]
+
+    @property
+    def hw_info(self) -> str:
+        """Hardware info."""
+        return self.data["hw_info"]
+
+    @property
+    def sw_info(self) -> str:
+        """SoftWare info."""
+        return self.data["sw_info"]
+
+    @property
+    def hepa_hours_left(self) -> timedelta:
+        """HYPA left hours."""
+        return timedelta(self.data["hypa_hours"])
+
+    @property
+    def hepa_life_left(self) -> int:
+        """HYPA left life percent."""
+        return self.data["hypa_life"]
+
+    @property
+    def charging(self) -> bool:
+        """FIXME: True if device is charging?"""
+        return bool(self.data["is_charge"])
+
+    @property
+    def working(self) -> bool:
+        """FIXME: True if device is working?"""
+        return bool(self.data["is_work"])
+
+    @property
+    def light_state(self) -> bool:
+        """FIXME: True if device ?"""
+        return bool(self.data["light_state"])
+
+    @property
+    def main_brush_hours_left(self) -> timedelta:
+        """Main brush left hours."""
+        return timedelta(self.data["main_brush_hours"])
```

Comment:
The consumable states are already exposed using `ViomiConsumableStatus`, so I'd simply remove the duplicate values (I think hepa is the only one not available through it, currently?)

Reply:
I what thinking replacing the `get_consumables` handler byt `get_properties`  with the props:
* "main_brush_hours",
* "main_brush_life",
* ...

It seems simpler to manage (no calculation/timedelta manipulation)
@rytilahti what do you think ?

Reply:
It is better to use the consumable status container for consistency; at some point someone may want to add support for non-rockrobo vacuums to homeassistant integration and having a common API makes it simpler to achieve.

Reply:
done